src/Sarif.Viewer.VisualStudio/LoadSarifLogService.cs

```diff
@@ -2,22 +2,46 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.Collections.Generic;
+using System.Linq;
 using Microsoft.CodeAnalysis.Sarif.Converters;
 using Microsoft.Sarif.Viewer.ErrorList; 
 
 namespace Microsoft.Sarif.Viewer
 {
-    public class LoadSarifLogService : SLoadSarifLogService, ILoadSarifLogService
+    public class LoadSarifLogService : SLoadSarifLogService, ILoadSarifLogService, ILoadSarifLogService2
     {
+        public void LoadSarifLog(string path, bool promptOnSchemaUpgrade = true)
+        {
+            if (string.IsNullOrWhiteSpace(path))
+            {
+                return;
+            }
+
+            try
+            {
+                ErrorListService.ProcessLogFile(path, SarifViewerPackage.Dte.Solution, ToolFormat.None, promptOnSchemaUpgrade);
+            }
+            catch (InvalidCastException) { }
+        }
+
         public void LoadSarifLog(string path)
         {
-            if (!string.IsNullOrWhiteSpace(path))
+            ErrorListService.ProcessLogFile(path, SarifViewerPackage.Dte.Solution, ToolFormat.None, promptOnLogConversions: true);
+        }
+
+        public void LoadSarifLogs(IEnumerable<string> paths) => this.LoadSarifLogs(paths, promptOnSchemaUpgrade: false);
+
+        public void LoadSarifLogs(IEnumerable<string> paths, bool promptOnSchemaUpgrade)
+        {
+            if (!paths.Any())
+            {
+                return;
+            }
+
+            foreach (var path in paths)
```

Comment:
>var [](start = 21, length = 3)

style nit, use 'string' here explicitly as opposed to 'var' keyword. Just for readability.

Reply:
Yep, we have a different style on our side, sorry for the bleed through. Fixed.

---
In reply to: [440862239](https://github.com/microsoft/sarif-visualstudio-extension/pull/143#discussion_r440862239) [](ancestors = 440862239)